### ISSUE TITLE
Testing accuracy

### ISSUE THREAD

**Issue by chokkarapu** [cardwing/Codes-for-Lane-Detection#16]
Hi,

You mentioned that during testing phase, to compute the accuracy and F2 related measure need to use original SCNN implementation.

In your training process, there were IOU computation and accuracy, can we utilize the same thing for testing purpose just like you used during the validation phase??.

These is the one im doing duirbg testing phase
![image](https://user-images.githubusercontent.com/3630515/49781637-81f3c580-fd39-11e8-9093-e12660d0001a.png)

and also 
![image](https://user-images.githubusercontent.com/3630515/49781670-9f289400-fd39-11e8-8868-1e1aebc16017.png)

And also 
![image](https://user-images.githubusercontent.com/3630515/49781694-b4052780-fd39-11e8-8f02-0daefc869158.png)

I computed, IoU_1, IoU_2, IoU_3, IoU_4 as per training code....

Look forward for your valuable response.

Thanks,
Anil.

**Comment by cardwing** [comment]
@chokkarapu, the IoU calculation used in validation is only to roughly check the performance of the model. However, the IoU calculation used in testing assumes that the lane width of ground-truth lanes and predicted lanes should be equal. Since the lane detection model can not output lane segments exactly like the labels, we have to first extract lane points from the probability maps, then using a fixed width lines to connect these points and finally calculate IoU and judge whether the lanes are correctly detected.

**Comment by chokkarapu** [comment]
> @chokkarapu, the IoU calculation used in validation is intended to roughly check the performance of the model. However, the IoU calculation used in testing assumes that the lane width of ground-truth lanes and predicted lanes should be equal. Since the lane detection model can not output lane segments exactly like the labels, we have to first extract lane points from the probability maps, then using a fixed width lines to connect these points and finally calculate IoU and judge whether the lanes are correctly detected. Otherwise, the IoU calculated in testing will be affected by the quality of generated lane segments and the width of ground-truth lanes.

@cardwing : So you mean to say, if we use the accuracy computed in validation cannot be directly used for testing purpose. If we use the accuracy will have diverse effect effect due to width of predicted lanes.
Is my understanding right?

**Comment by cardwing** [comment]
@chokkarapu, yes. ^_^
